esp/esdllib/esdl_script.cpp

```diff
@@ -769,155 +1046,145 @@ class CEsdlCustomTransform : public CInterfaceOf<IEsdlCustomTransform>
         }
     }
 
-    void processTransform(IEspContext * context, IPropertyTree *tgtcfg, IEsdlDefService &srvdef, IEsdlDefMethod &mthdef, StringBuffer & content, IPropertyTree * bindingCfg) override
-    {
-        processServiceAndMethodTransforms({static_cast<IEsdlCustomTransform*>(this)}, context, tgtcfg, srvdef, mthdef, content, bindingCfg);
-    }
-    void processTransform(IEspContext * context, IPropertyTree *tgtcfg, const char *service, const char *method, const char* reqtype, StringBuffer & content, IPropertyTree * bindingCfg) override
+    void processTransform(IEsdlScriptContext * scriptCtx, const char *srcSection, const char *tgtSection) override
     {
-        processServiceAndMethodTransforms({static_cast<IEsdlCustomTransform*>(this)}, context, tgtcfg, service, method, reqtype, content, bindingCfg);
+        processServiceAndMethodTransforms(scriptCtx, {static_cast<IEsdlCustomTransform*>(this)}, srcSection, tgtSection);
     }
 };
 
-void processServiceAndMethodTransforms(std::initializer_list<IEsdlCustomTransform *> const &transforms, IEspContext * context, IPropertyTree *tgtcfg, const char *service, const char *method, const char* reqtype, StringBuffer & content, IPropertyTree * bindingCfg)
+void processServiceAndMethodTransforms(IEsdlScriptContext * scriptCtx, std::initializer_list<IEsdlCustomTransform *> const &transforms, const char *srcSection, const char *tgtSection)
 {
     LogLevel level = LogMin;
+    if (!scriptCtx)
+        return;
     if (!transforms.size())
         return;
-    if (tgtcfg)
-        level = (unsigned) tgtcfg->getPropInt("@traceLevel", level);
+    if (isEmptyString(srcSection)||isEmptyString(tgtSection))
+        return;
+    level = scriptCtx->getXPathInt64("target/*/@traceLevel", level);
```

Comment:
LogLevel is unsigned, not __int64. This should be generating a compiler warning for potential loss of data.

Reply:
how do we get away with less than 9,223,372,036,854,775,807 log levels?

Reply:
I didn't think implementing 128 bit integers was within the scope of this story, but if you think more options make sense...

Reply:
So the getXPathInt64 is so we can support other values that might require large ints.  But I'll add a specific cast here so the compiler wont care.  I don't want to add mulitple getInt functions if I don't have to.